packages/interface/src/components/JoinButton.tsx

```diff
@@ -249,26 +260,24 @@ export const JoinButton = (): JSX.Element => {
           variant={isRegistered === undefined || isLoading ? "disabled" : "primary"}
           onClick={handleZupassVerify}
         >
-          Prove you can join with Zupass
+          Prove you voted with Zupass
         </Button>
       </div>
     );
   }
 
-  if (isEligibleToVote && !isRegistered) {
+  if (isEligibleToVote && poapLink !== "") {
     return (
       <div>
-        {isLoading ? (
-          <Spinner className="mr-2 h-4 w-4" />
-        ) : (
-          <Button
-            className="mb-4 sm:mb-0"
-            variant={isRegistered === undefined ? "disabled" : "primary"}
-            onClick={handleSignup}
-          >
-            Sign up to vote
-          </Button>
-        )}
+        <Button
+          className="btn mb-4 sm:mb-0"
+          variant={isRegistered === undefined || isLoading ? "disabled" : "primary"}
+          onClick={() => {
+            window.open(poapLink, "_blank");
+          }}
```

Comment:
Do not use arrow functions in tsx render. It should be moved to a separate variable.